### PR TITLE
Revert "Use consistent imageRef during container startup"

### PR DESCRIPTION
**Pull request by tallclair**
Automated cherry pick of #78261 upstream release 1.14

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

This reverts commit 26e3c8674e66f0d10170d34f5445f0aed207387f from #76665

The change broke the detection of user ID.

**Does this PR introduce a user-facing change?**:
```release-note
Fix broken detection of non-root image user ID
```

/sig node
/assign @yujuhong 
/priority critical-urgent

### CODE REVIEW

**Comment by yujuhong** [comment]
Need to patch 1.13 as well.

**Comment by ialidzhikov** [comment]
> Need to patch 1.13 as well.

Automated cherry-pick for release-1.13 proposed - #78322.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

**Comment by verwilst** [comment]
What's keeping this from being merged?

**Comment by mattjmcnaughton** [comment]
> What's keeping this from being merged?

My understanding is all cherry-picks need approval from the branch maintainer, and my guess is just they haven't gotten to this cherry-pick yet :)

**Pull request merged**
